src/lib.rs

```diff
@@ -530,6 +565,96 @@ pub fn is_loadbalancer_enabled(input: &str) -> bool {
     input.trim().eq_ignore_ascii_case("true")
 }
 
+/// Parse timeout overrides from metadata
+///
+/// Parses a JSON-like string containing timeout overrides. Any timeout not specified
+/// will use the default value.
+///
+/// # Format
+///
+/// ```text
+/// {queue:0,connect:5000,client:60000,server:180000}
+/// ```
+///
+/// All parameters are optional. Values are in milliseconds.
+///
+/// # Supported Parameters
+///
+/// * `queue` - Time to wait in queue for a connection slot (0 = unlimited)
+/// * `connect` - Time to wait for a connection to be established to a backend
+/// * `client` - Client inactivity timeout
+/// * `server` - Server response timeout
+///
+/// # Arguments
+///
+/// * `input` - The timeout configuration string
+///
+/// # Returns
+///
+/// A TimeoutConfig with values from the input merged with defaults
+pub fn parse_timeouts(input: &str) -> TimeoutConfig {
+    let trimmed = input.trim();
+    if trimmed.is_empty() {
+        return TimeoutConfig::default();
+    }
+
+    // Remove the curly braces if present
+    let content = trimmed.trim_start_matches('{').trim_end_matches('}');
+    if content.is_empty() {
+        return TimeoutConfig::default();
+    }
+
+    let mut config = TimeoutConfig::default();
+
+    // Split by comma and parse each key:value pair
+    for pair in content.split(',') {
+        let parts: Vec<&str> = pair.splitn(2, ':').collect();
+        if parts.len() != 2 {
+            warn!("Invalid timeout parameter format: {}", pair);
+            continue;
+        }
+
+        let key = parts[0].trim();
+        let value = parts[1].trim();
+
+        match key {
+            "queue" => {
+                if let Ok(v) = value.parse::<u32>() {
+                    config.queue = v;
+                } else {
+                    warn!("Invalid queue timeout value: {}", value);
+                }
+            }
+            "connect" => {
+                if let Ok(v) = value.parse::<u32>() {
+                    config.connect = v;
+                } else {
+                    warn!("Invalid connect timeout value: {}", value);
+                }
+            }
+            "client" => {
+                if let Ok(v) = value.parse::<u32>() {
+                    config.client = v;
+                } else {
+                    warn!("Invalid client timeout value: {}", value);
+                }
+            }
+            "server" => {
+                if let Ok(v) = value.parse::<u32>() {
+                    config.server = v;
+                } else {
+                    warn!("Invalid server timeout value: {}", value);
+                }
+            }
+            _ => {
+                warn!("Unknown timeout parameter: {}", key);
+            }
+        }
+    }
+
```

Comment:
Also, do we need sanity-checks on parameter values?  They're all unsigned-32-bit-capped, but is 4billion of any of these allowable?